src/compiler/reverse.jl

```diff
@@ -28,23 +28,38 @@ unwrapquote(x::QuoteNode) = x.value
 
 is_getproperty(ex) = iscall(ex, Base, :getproperty)
 
+# Allows us to resolve constants which have been stored in Variables.
+# e.g. `%1 = 1; %2 = %1``, or `%1 = identity; %1(...)`.
+trylookup(ir::IR, @nospecialize(v)) = v
+trylookup(ir::IR, v::Variable) = haskey(ir, v) ? trylookup(ir, ir[v].expr) : v
+# Only resolve GlobalRefs to traverse module hierarchies
+function trylookup(ir::IR, ref::GlobalRef)
+  isconst(ref.mod, ref.name) || return ref
+  val = getproperty(ref.mod, ref.name)
+  return val isa Module ? val : ref
+end
```

Comment:
Because this is probably going to be type unstable, you *may* want to write this as something like
```julia
function trylookup(ir::IR, v) # maybe @nospecialize(v) ?
    while true
        if v isa Variable
            v = ir[v].expr
        elseif v isa GlobalRef
           isconst(ref.mod, ref.name) || return ref
           val = getproperty(ref.mod, ref.name)
           return val isa Module ? val : ref
        else
           return v
        end
    end
end
```

Reply:
Thanks Mason. I remember thinking about a loop and rejecting it, but I can't think of why because the snippet you posted looks just fine. Will try incorporating it when I return to this PR.